runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -1776,18 +1774,16 @@ MM_CopyForwardScheme::clearCache(MM_EnvironmentVLHGC *env, MM_CopyScanCacheVLHGC
 			(discardSize <= ((uintptr_t)compactGroupForMarkData->_TLHRemainderTop - (uintptr_t)compactGroupForMarkData->_TLHRemainderBase))) {
 			/* Abandon the current entry in the cache */
 			env->_cycleState->_activeSubSpace->abandonHeapChunk(cache->cacheAlloc, cache->cacheTop);
-			discardHeapChunk(compactGroupForMarkData, cache->cacheAlloc, cache->cacheTop);
+			MM_HeapRegionDescriptorVLHGC *region = (MM_HeapRegionDescriptorVLHGC*)_extensions->heapRegionManager->tableDescriptorForAddress(cache->cacheAlloc);
+			compactGroupForMarkData->discardHeapChunk(cache->cacheAlloc, cache->cacheTop, region);
 		} else {
+			/* Abandon the current TLHRemainder if one exists */
 			if (NULL != compactGroupForMarkData->_TLHRemainderBase) {
-				/* Abandon the current entry in the cache */
+				MM_HeapRegionDescriptorVLHGC *region = (MM_HeapRegionDescriptorVLHGC*)_extensions->heapRegionManager->tableDescriptorForAddress(compactGroupForMarkData->_TLHRemainderBase);
 				env->_cycleState->_activeSubSpace->abandonHeapChunk(compactGroupForMarkData->_TLHRemainderBase, compactGroupForMarkData->_TLHRemainderTop);
-				discardHeapChunk(compactGroupForMarkData, compactGroupForMarkData->_TLHRemainderBase, compactGroupForMarkData->_TLHRemainderTop);
-				compactGroupForMarkData->resetTLHRemainder();
+				compactGroupForMarkData->discardHeapChunk(compactGroupForMarkData->_TLHRemainderBase, compactGroupForMarkData->_TLHRemainderTop, region);
 			}
```

Comment:
could we generalize discardTLHRemainder in CompactGroup, by adding boundary addresses, and use it here for both cases (current cache and current remainder)?

Reply:
or if too messy (the method name would not apply for current cache name), at least use it for current remainder case